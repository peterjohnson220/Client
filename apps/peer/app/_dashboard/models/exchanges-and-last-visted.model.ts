import { ExchangeListItem, generateMockExchangeListItem } from 'libs/models/peer';

export interface ExchangesAndLastVisted {
  Exchanges: ExchangeListItem[];
  LastVisitedExchangeId: number;
}

export function generateMockExchangesAndLastVisted(): ExchangesAndLastVisted {
  return {
    Exchanges: [generateMockExchangeListItem()],
    LastVisitedExchangeId: 10
  };
}
