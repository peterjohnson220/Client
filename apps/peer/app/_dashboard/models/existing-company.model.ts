export interface ExistingCompany {
  CompanyId: number;
  CompanyName: string;
  Industry: string;
  Size: number;
  FTEs: number;
  InExchange: boolean;
  PendingInvitation: boolean;
  Peers: string[];
}

export function generateMockExistingCompany(): ExistingCompany {
  return {
    CompanyId: 0,
    CompanyName: 'MockCompany',
    Industry: 'MockIndustry',
    Size: 0,
    FTEs: 0,
    InExchange: false,
    PendingInvitation: false,
    Peers: ['MockPeer']
  };
}
