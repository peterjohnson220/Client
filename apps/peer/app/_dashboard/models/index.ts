export * from './existing-company.model';
export * from './exchanges-and-last-visted.model';
