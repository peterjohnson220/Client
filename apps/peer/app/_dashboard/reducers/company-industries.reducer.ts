import { createEntityAdapter, EntityAdapter, EntityState } from '@ngrx/entity';

import * as fromCompanyIndustriesActions from '../actions/company-industries.actions';

export interface State extends EntityState<string> {
  loading: boolean;
  loadingError: boolean;
}

export const adapter: EntityAdapter<string> = createEntityAdapter<string>({
  selectId: x => x,
  sortComparer: (a, b) => {
    if (a < b) {
      return -1;
    }
    if (a > b) {
      return 1;
    }
    return 0;
  }
});

const initialState: State = adapter.getInitialState({
  loading: false,
  loadingError: false
});

// Reducer function
export function reducer(state = initialState, action: fromCompanyIndustriesActions.Actions): State {
  switch (action.type) {
    case fromCompanyIndustriesActions.LOAD_COMPANY_INDUSTRIES: {
      return {
        ...adapter.removeAll(state),
        loading: true,
        loadingError: false
      };
    }
    case fromCompanyIndustriesActions.LOAD_COMPANY_INDUSTRIES_SUCCESS: {
      const exchangeJobMappings: string[] = action.payload;
      return {
        ...adapter.addAll(exchangeJobMappings, state),
        loading: false
      };
    }
    case fromCompanyIndustriesActions.LOAD_COMPANY_INDUSTRIES_ERROR: {
      return {
        ...state,
        loading: false,
        loadingError: true
      };
    }
    default: {
      return state;
    }
  }
}

// Selector functions
export const getLoading = (state: State) => state.loading;
export const getLoadingError = (state: State) => state.loadingError;
