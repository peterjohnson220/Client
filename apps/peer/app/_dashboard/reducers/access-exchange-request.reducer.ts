import { EntityAdapter, createEntityAdapter } from '@ngrx/entity';

import { ExchangeRequestTypeEnum, AvailableExchangeItem } from 'libs/models/peer';

import { ExchangeRequestEntityState, createExchangeRequestReducer } from '../../shared/reducers/exchange-request.reducer';

export interface State extends ExchangeRequestEntityState<AvailableExchangeItem> { }

export const adapter: EntityAdapter<AvailableExchangeItem> = createEntityAdapter<AvailableExchangeItem>({
  selectId: (availableExchange: AvailableExchangeItem) => availableExchange.ExchangeId
});

// Reducer
export function reducer(state, action) {
  return createExchangeRequestReducer<AvailableExchangeItem> (
    ExchangeRequestTypeEnum.Access,
    adapter
  )(state, action);
}

// Selector functions
export const getLoadingRequestContext = (state: State) => {
  return {
    Query: state.searchTerm,
    CompanyFilterId: state.filterOptions ? state.filterOptions.companyFilterId : null
  };
};
