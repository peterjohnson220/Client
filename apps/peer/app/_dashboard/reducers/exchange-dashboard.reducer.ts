import {ChartItem, ExchangeJobComparison} from 'libs/models';
import {ExchangeChartTypeEnum} from 'libs/models/peer/exchange-chart-type.enum';

import * as fromExchangeDashboardActions from '../actions/exchange-dashboard.actions';

export interface State {
  loadingCompanyChart: boolean;
  loadingCompanyChartError: boolean;
  companyChartItems: ChartItem[];
  loadingJobChart: boolean;
  loadingJobChartError: boolean;
  jobChartItems: ChartItem[];
  loadingIndustryChart: boolean;
  loadingIndustryChartError: boolean;
  industryChartItems: ChartItem[];
  loadingJobFamilyChart: boolean;
  loadingJobFamilyChartError: boolean;
  jobFamilyChartItems: ChartItem[];
  loadingRevenueChart: boolean;
  loadingRevenueChartError: boolean;
  revenueChartItems: ChartItem[];
  detailChartCategory: string;
  detailChartType: string;
  loadingDetailChart: boolean;
  loadingDetailChartError: boolean;
  detailChartItems: ChartItem[];
  sidebarVisible: boolean;
  mapHasDataItem: boolean;
  loadingMapHasData: boolean;
  loadingMapHasDataError: boolean;
  }

// Initial State
export const initialState: State = {
  loadingCompanyChart: false,
  loadingCompanyChartError: false,
  companyChartItems: null,
  loadingJobChart: false,
  loadingJobChartError: false,
  jobChartItems: null,
  loadingIndustryChart: false,
  loadingIndustryChartError: false,
  industryChartItems: null,
  loadingJobFamilyChart: false,
  loadingJobFamilyChartError: false,
  jobFamilyChartItems: null,
  loadingRevenueChart: false,
  loadingRevenueChartError: false,
  revenueChartItems: null,
  detailChartCategory: null,
  detailChartType: null,
  loadingDetailChart: false,
  loadingDetailChartError: false,
  detailChartItems: null,
  sidebarVisible: false,
  mapHasDataItem: false,
  loadingMapHasData: false,
  loadingMapHasDataError: false
  };

// Reducer
export function reducer(
  state = initialState,
  action: fromExchangeDashboardActions.Actions
): State {
  switch (action.type) {
    case fromExchangeDashboardActions.LOAD_COMPANY_CHART: {
      return {
        ...state,
        companyChartItems: null,
        loadingCompanyChart: true,
        loadingCompanyChartError: false
      };
    }
    case fromExchangeDashboardActions.LOAD_COMPANY_CHART_SUCCESS: {
      return {
        ...state,
        companyChartItems: action.payload,
        loadingCompanyChart: false,
        loadingCompanyChartError: false
      };
    }
    case fromExchangeDashboardActions.LOAD_COMPANY_CHART_ERROR: {
      return {
        ...state,
        loadingCompanyChart: false,
        loadingCompanyChartError: true
      };
    }
    case fromExchangeDashboardActions.LOAD_JOB_CHART: {
      return {
        ...state,
        jobChartItems: null,
        loadingJobChart: true,
        loadingJobChartError: false
      };
    }
    case fromExchangeDashboardActions.LOAD_JOB_CHART_SUCCESS: {
      return {
        ...state,
        jobChartItems: action.payload,
        loadingJobChart: false,
        loadingJobChartError: false
      };
    }
    case fromExchangeDashboardActions.LOAD_JOB_CHART_ERROR: {
      return {
        ...state,
        loadingJobChart: false,
        loadingJobChartError: true
      };
    }
    case fromExchangeDashboardActions.LOAD_INDUSTRY_CHART: {
      return {
        ...state,
        industryChartItems: null,
        loadingIndustryChart: true,
        loadingIndustryChartError: false
      };
    }
    case fromExchangeDashboardActions.LOAD_INDUSTRY_CHART_SUCCESS: {
      return {
        ...state,
        industryChartItems: action.payload,
        loadingIndustryChart: false,
        loadingIndustryChartError: false
      };
    }
    case fromExchangeDashboardActions.LOAD_INDUSTRY_CHART_ERROR: {
      return {
        ...state,
        loadingIndustryChart: false,
        loadingIndustryChartError: true
      };
    }
    case fromExchangeDashboardActions.LOAD_JOB_FAMILY_CHART: {
      return {
        ...state,
        jobFamilyChartItems: null,
        loadingJobFamilyChart: true,
        loadingJobFamilyChartError: false
      };
    }
    case fromExchangeDashboardActions.LOAD_JOB_FAMILY_CHART_SUCCESS: {
      return {
        ...state,
        jobFamilyChartItems: action.payload,
        loadingJobFamilyChart: false,
        loadingJobFamilyChartError: false
      };
    }
    case fromExchangeDashboardActions.LOAD_JOB_FAMILY_CHART_ERROR: {
      return {
        ...state,
        loadingJobFamilyChart: false,
        loadingJobFamilyChartError: true
      };
    }
    case fromExchangeDashboardActions.LOAD_REVENUE_CHART: {
      return {
        ...state,
        revenueChartItems: null,
        loadingRevenueChart: true,
        loadingRevenueChartError: false
      };
    }
    case fromExchangeDashboardActions.LOAD_REVENUE_CHART_SUCCESS: {
      return {
        ...state,
        revenueChartItems: action.payload,
        loadingRevenueChart: false,
        loadingRevenueChartError: false
      };
    }
    case fromExchangeDashboardActions.LOAD_REVENUE_CHART_ERROR: {
      return {
        ...state,
        loadingRevenueChart: false,
        loadingRevenueChartError: true
      };
    }
    case fromExchangeDashboardActions.LOAD_DETAIL_CHART: {
      return {
        ...state,
        detailChartType: action.payload.ChartType,
        detailChartCategory: action.payload.Category,
        sidebarVisible: true,
        detailChartItems: null,
        loadingDetailChart: true,
        loadingDetailChartError: false
      };
    }
    case fromExchangeDashboardActions.LOAD_DETAIL_CHART_SUCCESS: {
      return {
        ...state,
        detailChartItems: action.payload,
        loadingDetailChart: false,
        loadingDetailChartError: false
      };
    }
    case fromExchangeDashboardActions.LOAD_DETAIL_CHART_ERROR: {
      return {
        ...state,
        loadingDetailChart: false,
        loadingDetailChartError: true
      };
    }
    case fromExchangeDashboardActions.CLOSE_SIDEBAR: {
      return {
        ...state,
        sidebarVisible: false,
        detailChartItems: null,
        loadingDetailChart: false,
        loadingDetailChartError: false
      };
    }
    case fromExchangeDashboardActions.LOAD_MAP_COUNT: {
      return {
        ...state,
        mapHasDataItem: false,
        loadingMapHasData: true,
        loadingMapHasDataError: false
      };
    }
    case fromExchangeDashboardActions.LOAD_MAP_COUNT_SUCCESS: {
      return {
        ...state,
        mapHasDataItem: action.payload,
        loadingMapHasData: false,
        loadingMapHasDataError: false
      };
    }
    case fromExchangeDashboardActions.LOAD_MAP_COUNT_ERROR: {
      return {
        ...state,
        mapHasDataItem : false,
        loadingMapHasData: false,
        loadingMapHasDataError: true
      };
    }
    case fromExchangeDashboardActions.LOAD_EXCHANGE_JOB_ORGS: {
      const payload: ExchangeJobComparison = action.payload.selectedExchangeJobComparison;
      return {
        ...state,
        detailChartType: ExchangeChartTypeEnum.ExchangeJobOrgs,
        detailChartCategory: payload.ExchangeJobTitle,
        sidebarVisible: true,
        detailChartItems: null,
        loadingDetailChart: true,
        loadingDetailChartError: false
      };
    }
    case fromExchangeDashboardActions.LOAD_EXCHANGE_JOB_ORGS_SUCCESS: {
      return {
        ...state,
        detailChartItems: action.payload,
        loadingDetailChart: false,
        loadingDetailChartError: false
      };
    }
    case fromExchangeDashboardActions.LOAD_EXCHANGE_JOB_ORGS_ERROR: {
      return {
        ...state,
        loadingDetailChart: false,
        loadingDetailChartError: true
      };
    }
    default: {
      return state;
    }
  }
}

// Selector Functions
export const getCompanyChartItems = (state: State) => state.companyChartItems;
export const getLoadingCompanyChart = (state: State) => state.loadingCompanyChart;
export const getLoadingCompanyChartError = (state: State) => state.loadingCompanyChartError;
export const getJobChartItems = (state: State) => state.jobChartItems;
export const getLoadingJobChart = (state: State) => state.loadingJobChart;
export const getLoadingJobChartError = (state: State) => state.loadingJobChartError;
export const getIndustryChartItems = (state: State) => state.industryChartItems;
export const getLoadingIndustryChart = (state: State) => state.loadingIndustryChart;
export const getLoadingIndustryChartError = (state: State) => state.loadingIndustryChartError;
export const getJobFamilyChartItems = (state: State) => state.jobFamilyChartItems;
export const getLoadingJobFamilyChart = (state: State) => state.loadingJobFamilyChart;
export const getLoadingJobFamilyChartError = (state: State) => state.loadingJobFamilyChartError;
export const getRevenueChartItems = (state: State) => state.revenueChartItems;
export const getLoadingRevenueChart = (state: State) => state.loadingRevenueChart;
export const getLoadingRevenueChartError = (state: State) => state.loadingRevenueChartError;
export const getDetailChartType = (state: State) => state.detailChartType;
export const getDetailChartCategory = (state: State) => state.detailChartCategory;
export const getDetailChartItems = (state: State) => state.detailChartItems;
export const getLoadingDetailChart = (state: State) => state.loadingDetailChart;
export const getLoadingDetailChartError = (state: State) => state.loadingDetailChartError;
export const getSidebarVisible = (state: State) => state.sidebarVisible;
export const getMapHasDataItem = (state: State) => state.mapHasDataItem;
export const getLoadingMapHasData = (state: State) => state.loadingMapHasData;
export const getLoadingMapHasDataError = (state: State) => state.loadingMapHasDataError;
