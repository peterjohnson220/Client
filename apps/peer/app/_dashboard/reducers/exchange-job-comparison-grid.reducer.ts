import { createEntityAdapter, EntityAdapter, EntityState } from '@ngrx/entity';

import { ExchangeJobComparison } from 'libs/models/peer';
import { createGridReducer } from 'libs/core/reducers/grid.reducer';
import { GridTypeEnum } from 'libs/models/common';

import * as fromExchangeJobComparisonGridActions from '../actions/exchange-job-comparison-grid.actions';

export interface State extends EntityState<ExchangeJobComparison> {
  loading: boolean;
  loadingError: boolean;
  total: number;
  loadingExchangeJobOrgs: boolean;
  loadingExchangeJobOrgsError: boolean;
  exchangeJobOrgs: string[];
}

export const adapter: EntityAdapter<ExchangeJobComparison> = createEntityAdapter<ExchangeJobComparison>({
  selectId: (exchangeJobComparison: ExchangeJobComparison) => exchangeJobComparison.ExchangeJobToCompanyJobId
});

const initialState: State = adapter.getInitialState({
  loading: false,
  loadingError: false,
  total: 0,
  loadingExchangeJobOrgs: false,
  loadingExchangeJobOrgsError: false,
  exchangeJobOrgs: []
});


// Reducer function
export function reducer(state, action) {
  return createGridReducer(
    GridTypeEnum.ExchangeJobComparison,
    (featureState = initialState,  featureAction: fromExchangeJobComparisonGridActions.Actions): State => {
      switch (featureAction.type) {
        case fromExchangeJobComparisonGridActions.LOAD_EXCHANGE_JOB_COMPARISONS: {
          return {
            ...featureState,
            loading: true,
            loadingError: false
          };
        }
        case fromExchangeJobComparisonGridActions.LOAD_EXCHANGE_JOB_COMPARISONS_SUCCESS: {
          const exchangeJobMappings: ExchangeJobComparison[] = featureAction.payload.data;
          return {
            ...adapter.removeAll(featureState),
            ...adapter.setAll(exchangeJobMappings, featureState),
            total: featureAction.payload.total,
            loading: false
          };
        }
        case fromExchangeJobComparisonGridActions.LOAD_EXCHANGE_JOB_COMPARISONS_ERROR: {
          return {
            ...featureState,
            loading: false,
            loadingError: true
          };
        }
        default: {
          return featureState;
        }
      }
    }, {
      take: 10
    })(state, action);
}

// Selector functions
export const getLoading = (state: State) => state.loading;
export const getLoadingError = (state: State) => state.loadingError;
export const getTotal = (state: State) => state.total;
