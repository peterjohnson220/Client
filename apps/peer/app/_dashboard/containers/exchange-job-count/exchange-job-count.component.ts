import { Component, OnInit, OnDestroy } from '@angular/core';
import { ActivatedRoute, ParamMap, Router } from '@angular/router';

import { Store } from '@ngrx/store';
import { Observable, Subscription } from 'rxjs';

import { Exchange, GetChartRequest, ChartItem, ExchangeChartTypeEnum } from 'libs/models';

import * as fromPeerDashboardReducer from '../../reducers';
import * as fromSharedPeerReducer from '../../../shared/reducers';
import * as fromExchangeDashboardActions from '../../actions/exchange-dashboard.actions';

@Component({
  selector: 'pf-exchange-job-count',
  templateUrl: './exchange-job-count.component.html',
  styleUrls: ['./exchange-job-count.component.scss']
})

export class ExchangeJobCountComponent implements OnInit, OnDestroy {
  exchangeId: number;
  chartItem: ChartItem;
  jobChartItems$: Observable<ChartItem[]>;
  jobChartItemsSubscription: Subscription;
  loadingJobChartItems$: Observable<boolean>;
  loadingJobChartItemsError$: Observable<boolean>;

  constructor(
    private store: Store<fromPeerDashboardReducer.State>,
    private route: ActivatedRoute
  ) {
    this.jobChartItems$ = this.store.select(fromPeerDashboardReducer.getExchangeDashboardJobChartItems);
    this.loadingJobChartItems$ = this.store.select(fromPeerDashboardReducer.getExchangeDashboardLoadingJobChart);
    this.loadingJobChartItemsError$ = this.store.select(fromPeerDashboardReducer.getExchangeDashboardLoadingJobChartError);
    this.route.paramMap.subscribe((params: ParamMap) => this.exchangeId = +params.get('id'));
  }

  ngOnInit() {
    this.jobChartItemsSubscription = this.jobChartItems$.subscribe(jobChartItem => {
      if (jobChartItem) {
        this.chartItem = jobChartItem[0];
      }
    });
  }

  ngOnDestroy() {
    this.jobChartItemsSubscription.unsubscribe();
  }
}




