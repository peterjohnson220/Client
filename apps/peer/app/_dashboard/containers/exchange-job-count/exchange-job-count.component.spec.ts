import { NO_ERRORS_SCHEMA } from '@angular/core';
import { TestBed, ComponentFixture } from '@angular/core/testing';
import { ActivatedRoute } from '@angular/router';

import { StoreModule, Store, combineReducers } from '@ngrx/store';
import spyOn = jest.spyOn;

import * as fromRootState from 'libs/state/state';
import { generateMockChartItem } from 'libs/models';
import { ActivatedRouteStub } from 'libs/test/activated-route-stub';

import * as fromPeerDashboardReducer from '../../reducers';
import { ExchangeJobCountComponent } from './exchange-job-count.component';

describe('Peer Dashboard - Exchange Job Count', () => {
  let fixture: ComponentFixture<ExchangeJobCountComponent>;
  let instance: ExchangeJobCountComponent;
  let store: Store<fromPeerDashboardReducer.State>;
  let route: any;

  // Configure Testing Module for before each test
  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [
        StoreModule.forRoot({
          ...fromRootState.reducers,
          peer_dashboard: combineReducers(fromPeerDashboardReducer.reducers)
        }),
      ],
      providers: [
        {
          provide: ActivatedRoute,
          useValue: new ActivatedRouteStub(),
        }
      ],
      declarations: [
        ExchangeJobCountComponent
      ],
      // Shallow Testing
      schemas: [ NO_ERRORS_SCHEMA ]
    });

    store = TestBed.inject(Store);
    route = TestBed.inject(ActivatedRoute);

    route.setParamMap({ id: 1 });

    fixture = TestBed.createComponent(ExchangeJobCountComponent);
    instance = fixture.componentInstance;

    spyOn(store, 'dispatch');
  });

  it('should display the job count', () => {
    instance.chartItem = { ...generateMockChartItem(), Value: 10 };

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
