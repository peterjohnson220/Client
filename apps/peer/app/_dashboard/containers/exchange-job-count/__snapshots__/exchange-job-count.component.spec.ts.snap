// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Peer Dashboard - Exchange Job Count should display the job count 1`] = `
<pf-exchange-job-count
  chartItem={[Function Object]}
  exchangeId={[Function Number]}
  jobChartItems$={[Function Store]}
  jobChartItemsSubscription={[Function Subscriber]}
  loadingJobChartItems$={[Function Store]}
  loadingJobChartItemsError$={[Function Store]}
  route={[Function ActivatedRouteStub]}
  store={[Function Store]}
>
  <h1
    class="job-count"
  >
    
    10
  
  </h1>
  <h3>
    
    Category
  
  </h3>
  
</pf-exchange-job-count>
`;
